Examples/MAX32650/OTP_Dump/main.c

```diff
@@ -0,0 +1,164 @@
+ *
+ ******************************************************************************/
+
+/**
+ * @file    main.c
+ * @brief   OTP Dump Example
+ * @details This example demonstrates how to read/write data from OTP memory
+ *
+ */
+
+/***** Includes *****/
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "flc.h"
+#include "mxc_device.h"
+
+/***** Definitions *****/
+/*
+ *  Attention if you set WITH_WRITE_TEST flag
+ *  This example will search OTP and try to find 0xffffffff data
+ *  If it could be found 4 test bytes will be written in that address.
+ */
+#define WITH_WRITE_TEST 0 // set it to test write test
+
+// OTP Address
+#define OTP_MAXIM_AREA MXC_INFO_MEM_BASE
+#define OTP_MAXIM_AREA_SIZE 0x1000
+
+#define OTP_USER_AREA (OTP_MAXIM_AREA + OTP_MAXIM_AREA_SIZE)
+#define OTP_USER_AREA_SIZE (MXC_INFO_MEM_SIZE - OTP_MAXIM_AREA_SIZE)
+
+/***** Static Functions *****/
+static void dump_section(unsigned int address, unsigned int length)
+{
+    unsigned int i;
+    volatile uint32_t *addr = (uint32_t *)address;
+
+    // unlock otp to access it
+    MXC_FLC_UnlockInfoBlock((uint32_t)address);
+
+    length /= 4; // on each loop print 4 bytes
+
+    for (i = 0; i < length; i++) {
+        if (!(i % 4)) {
+            printf("\n0x%08x:", (unsigned int)addr);
+        }
+
+        // add extra space
+        if (!(i % 2)) {
+            printf("   ");
+        }
+
+        printf(" %08x", *addr);
+        addr++;
+    }
+
+    // lock otp
+    MXC_FLC_LockInfoBlock((uint32_t)address);
+}
+
+#if WITH_WRITE_TEST
+static int write_test(void)
+{
+    int ret = 0;
+    uint32_t test_val = 0x11223344;
+    volatile uint32_t *addr = (uint32_t *)OTP_USER_AREA;
+    volatile uint32_t *end_addr = (uint32_t *)(OTP_USER_AREA + 1024);
+
+    // unlock otp
+    MXC_FLC_UnlockInfoBlock(OTP_USER_AREA);
+
+    // find free slot
+    while (addr < end_addr) {
+        if (*addr == 0xffffffff) {
+            printf("\n\nFree Addr: 0x%X\n", (uint32_t)addr);
+            break;
+        }
+
+        addr++;
+    }
+
+    if (addr >= end_addr) {
+        ret = -1; // means free slot not found
+        printf("\nFree Index Not Found in OTP\n");
+    }
+
+    if (ret == 0) {
+        ret = MXC_FLC_Write32((uint32_t)addr, test_val);
+
+        if (ret) {
+            printf("FLC Write Error: %d\n", ret);
+        }
+    }
+
+    // lock otp
+    MXC_FLC_LockInfoBlock(OTP_USER_AREA);
+
+    if (ret == 0) {
+        /* Dump user section */
+        printf("\n\n***** After Write OTP Section *****\n");
+        dump_section((unsigned int)addr, 32);
+    }
+
+    return ret;
+}
+#endif
+
+//******************************************************************************
+int main(void)
+{
+    printf("\n\n***** OTP Memory Read/Write Example *****\n");
+    printf("***** This example demonstrates how you can read/write OTP memory *****\n");
+
+    /* Dump maxim section */
+    printf("\n\n***** MAXIM AREA *****\n");
+    dump_section(OTP_MAXIM_AREA, 1024);
```

Comment:
Ditto

"Update to "ADI AREA" instead? If applicable, we should probably start replacing the "Maxim" names with "ADI" or something similar."